core/arch/arch.c

```diff
@@ -3273,7 +3273,14 @@ check_syscall_method(dcontext_t *dcontext, instr_t *instr)
         ASSERT(get_syscall_method() == SYSCALL_METHOD_UNINITIALIZED ||
                get_syscall_method() == SYSCALL_METHOD_INT);
 #    ifdef LINUX
-        if (new_method == SYSCALL_METHOD_SYSENTER) {
+
+        /* i#4407: An OP_syscall instruction on 32-bit AMD returns to a hardcoded vsyscall PC no
+         * matter where it is. Thus we must hook the vsyscall just like we do for OP_sysenter.
```

Comment:
Looks like it's failing clang-format b/c the line's too long so it has to wrap it:
```
-        /* i#4407: An OP_syscall instruction on 32-bit AMD returns to a hardcoded vsyscall PC no
-         * matter where it is. Thus we must hook the vsyscall just like we do for OP_sysenter.
+        /* i#4407: An OP_syscall instruction on 32-bit AMD returns to a hardcoded vsyscall
+         * PC no matter where it is. Thus we must hook the vsyscall just like we do for
+         * OP_sysenter.
```

Reply:
Fixed